### PR TITLE
Integrate Percy.io for visual regression tests

### PR DESCRIPTION
**Pull request by timhaines**
Hey Mike, thanks for the chat today.

When you sign up to [Percy](https://percy.io), it'll do a pretty good job of guiding you through the environment variable and github integration steps.

For convenience, here's a couple of links to **Percy Docs**:
- [TravisCI](https://percy.io/docs/setup/travis-ci)
- [Capybara](https://percy.io/docs/clients/ruby/capybara)

The second commit in this PR configures Travis to only run Percy against the latest Ruby.  We got this from a reliable source, but haven't tested it.  If this doesn't "just work", let me know and I'll get it sorted.

If you strike any issues at all, let us know.  I'm on DM, or you can Mike & I via Intercom/Email.

### CODE REVIEW

**Comment by timhaines** [comment]
@mperham Have seen this fail in Travis.  Going to sort that out now.

**Comment by timhaines** [comment]
@mperham Fixed the Travis issue, all good to merge now AFAIK.  👍

File: sidekiq.gemspec [hunk 1]
```diff
@@ -23,4 +23,10 @@ Gem::Specification.new do |gem|
   gem.add_development_dependency      'minitest', '~> 5.10', '>= 5.10.1'
   gem.add_development_dependency      'rake', '~> 10.0'
   gem.add_development_dependency      'rails', '>= 3.2.0'
+
+  gem.add_development_dependency      'capybara'
```
Comment (mperham) [inline sidekiq.gemspec]: Can you lock these down to the current major version?

File: test/test_web.rb [hunk 2]
```diff
@@ -34,24 +58,37 @@ def perform(a, b)
...
-      rackenv = {'HTTP_ACCEPT_LANGUAGE' => 'zh-tw'}
-      get '/', {}, rackenv
-      assert_match(/資訊主頁/, last_response.body)
-      rackenv = {'HTTP_ACCEPT_LANGUAGE' => 'nb'}
-      get '/', {}, rackenv
-      assert_match(/Oversikt/, last_response.body)
+      page.driver.headers = { 'Accept-Language' => 'ru,en' }
+      visit '/'
+      assert_text('Панель управления')
+      Percy::Capybara.snapshot(page, name: 'Dashboard (Russian)')
```
Comment (mperham) [inline test/test_web.rb]: Could you wrap this API in a helper method at the bottom?  Something like `snap(page, name: ...)` would be fine.  I want to make it trivial to remove Percy if necessary.

**Comment by timhaines** [comment]
@mperham have addressed your review comments.  Both great ideas, thanks.

**Comment by mperham** [comment]
I'd like to get to a place where PERCY_ENABLE=0 means no Percy gems are required and no Percy code is executed.  Possible?

**Comment by timhaines** [comment]
@mperham I've pushed a commit for PERCY_ENABLE=0 preventing the gem being required and Percy code from being executed.  Is that what you had in mind?

This is executed by already by 3 of the 4 configurations in Travis too.

**Comment by mperham** [comment]
Thank you Tim!

**Comment by mperham** [comment]
First error:

```
Cliver::Dependency::NotFound: Could not find an executable ["phantomjs"] on your path.
    /Users/mikeperham/.gem/ruby/2.3.2/gems/cliver-0.3.2/lib/cliver/dependency.rb:143:in `raise_not_found!'
    /Users/mikeperham/.gem/ruby/2.3.2/gems/cliver-0.3.2/lib/cliver/dependency.rb:116:in `detect!'
    /Users/mikeperham/.gem/ruby/2.3.2/gems/cliver-0.3.2/lib/cliver.rb:24:in `detect!'
    /Users/mikeperham/.gem/ruby/2.3.2/gems/poltergeist-1.12.0/lib/capybara/poltergeist/client.rb:48:in `initialize'
    /Users/mikeperham/.gem/ruby/2.3.2/gems/poltergeist-1.12.0/lib/capybara/poltergeist/client.rb:14:in `new'
```

**Comment by mperham** [comment]
Figured out that `brew install phantomjs` is required now to run the test suite.  The test suite has gone from 1 sec to 9 sec locally, presumably due to having to start a headless browser instance.

**Comment by timhaines** [comment]
@mperham Apologies for missing the PhantomJS note.  PhantomJS is used by Capybara & Poltergeist for the UI tests now, even when Percy's snapshots are disabled.

Are you seeing 9 seconds with the snapshots enabled or disabled?   On my 2 year old macbook, these tests moved from 0.75 seconds to 3.5 seconds with the snapshots disabled in the dev environment.  It's intended that you'd only run snapshots in CI, so this was the comparison I was paying attention to.  

With snapshots enabled, the test suite was taking 15 - 25 seconds for me.

**Pull request merged**